lib/remote/jsonrpcconnection-pki.cpp

```diff
@@ -289,6 +269,20 @@ void JsonRpcConnection::SendCertificateRequest(const JsonRpcConnection::Ptr& acl
 
 	/* Path is empty if this is our own request. */
 	if (path.IsEmpty()) {
+		if (aclient && !origin) {
+			Log(LogInformation, "JsonRpcConnection")
+				<< "Requesting new certificate for this Icinga instance from endpoint '" << aclient->GetIdentity() << "'.";
+
+			if (!aclient->m_SendCsrTimer) {
+				aclient->m_SendCsrTimer = new Timer();
+				aclient->m_SendCsrTimer->OnTimerExpired.connect([aclient](const Timer * const&) {
+					aclient->SendCertificateRequest(aclient, nullptr, "");
+				});
+				aclient->m_SendCsrTimer->SetInterval(29 * 24 * 60 * 60);
```

Comment:
Do we really have to push it to the last minute (or day in this case)?

Reply:
The operation is not that expensive so that we have to avoid doing it as long as we can. It's all about timing. The code ensures that the node always has a certificate with a remaining validity of RENEW_THRESHOLD - RENEW_INTERVAL. But only if everything is running properly. So with particularly bad timing, your node might crash 28.9 days after the last renewal attempt with a remaining certificate lifetime of 1.1 days. If it's not fixed within that time frame, manual intervention for the certificate is needed. Attempting a renewal every 10 days would trivially extend the time you have to fix the host before manual intervention is needed to 20 days.